src/databricks/labs/lsql/dashboards.py

```
@@ -893,6 +896,46 @@ def _from_dashboard_folder(cls, folder: Path) -> "DashboardMetadata":
             tiles.append(tile)
         return cls(display_name=folder.name, _tiles=tiles)
 
+    @staticmethod
+    def _add_to_zip(file_path: Path) -> None:
+        """Add a file to a ZIP archive, handling duplicates by renaming."""
+        zip_path = file_path.parent / "export_to_zipped_csv.zip"
+
+        try:
+            with ZipFile(zip_path, "a") as zip_file:
+                zip_file.write(file_path, arcname=file_path)
+
+        except FileNotFoundError:
+            logger.warning(f"File {file_path} not found.")
+        except PermissionError:
+            logger.warning(f"Permission denied for {file_path} or {zip_path}.")
```

```suggestion
```
don't swallow errors here.